cockroachdb/templates/statefulset.yaml

```diff
@@ -202,8 +202,8 @@ spec:
             {{- with .Values.conf.attrs }}
               --attrs={{ join `:` . }}
             {{- end }}
-              --http-port={{ index .Values.conf `http-port` | int64 }}
-              --port={{ .Values.conf.port | int64 }}
+              --http-port={{ index .Values.service.ports.http.port | int64 }}
```

Comment:
If someone has `conf.port` and/or `conf.http-port` set, this change will break the chart, right? Could we set them to `""` in `values.yaml` and only use them if they are set somewhere explicitly? Maybe something like.

```
{{- if .Values.conf.port }}
  --port={{ .Values.conf.port | int64 }} #WARNING this parameter is deprecated and will be removed in a future version. Use `.Values.service.ports.grpc.internal` instead
{{ - else }}
--port={{ .Values.service.ports.grpc.internal.port | int64 }}
{{ - end }}
```
wdyt?

Reply:
Yes, that would generally be true in most cases. However, for the Cockroach cluster to function correctly here, `.Values.conf.port` must match `.Values.service.ports.grpc.internal.port`, and similarly, `.Values.conf.http-port` must match `.Values.service.ports.http.port`.

For any cluster in a ready state, both sets of values should already be the same. But for added safety, I'm okay with keeping the old placeholders as a precaution.

Reply:
Added the changes.